### PR TITLE
Dotnet 8 Isolated Mariner Image

### PR DESCRIPTION
**Pull request by zahalzel**
Successfully builds for dotnet 7 mariner: 
https://github.com/Azure/azure-functions-docker/pull/1027/checks?check_run_id=21423485289

Successfully builds for dotnet 8 mariner:
https://github.com/Azure/azure-functions-docker/pull/1027/checks?check_run_id=21423663315

Releasing new Host versions.
[V3 Release](PASTE_V3_LINK_HERE)
[V2 Release](PASTE_V2_LINK_HERE)

---

### PR information

- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### CODE REVIEW

**Review by CooperLink — COMMENTED** [review]
This lgtm. We need to add this to our build pipelines though before we can merge

**Review by pragnagopa — DISMISSED** [review]
Please see:
1. https://github.com/Azure/azure-functions-docker/pull/918/files
2.https://github.com/Azure/azure-functions-docker/pull/896/files

for reference on build and publish pipeline updates

**Comment by zahalzel** [comment]
Waiting on this PR to merge: https://github.com/Azure/azure-functions-docker/pull/1036/files

File: host/4/dotnet-build.yml
Comment (zahalzel) [inline host/4/dotnet-build.yml]: Waiting on this PR: https://github.com/Azure/azure-functions-docker/pull/1036/files

**Review by CooperLink — DISMISSED** [review]

File: host/4/dotnet-build.yml
Comment (CooperLink) [inline host/4/dotnet-build.yml:80]: Both images are being tested using mariner now and succeed at those tests? Could you please paste a link to a successful build for both images

Comment (CooperLink) [reply host/4/dotnet-build.yml:80]: Also please check the rest of the repo to make sure nothing else is affected by the dotnet7 mariner image name change.  You shouldn't need to change any other pipelines but please double check. 

For example, make sure that the image produced by the build pipeline has the same exact name.  This is important to maintain release pipelines. If the image name changes it could break a downstream pipeline

File: host/4/republish.yml
Comment (CooperLink) [inline host/4/republish.yml:138]: Just make sure that this pipeline could still pick up the dotnet7 mariner image. IE - The output image name has not changed

**Review by CooperLink — COMMENTED** [review]
Please make sure that the Build Pipeline change for dotnet7 mariner doesn't change the image name created by the pipeline.

File: host/4/republish.yml
Comment (zahalzel) [reply host/4/republish.yml:138]: Yep, this change only affects the path to the file and the display name for the tests: image

File: host/4/dotnet-build.yml
Comment (zahalzel) [reply host/4/dotnet-build.yml:80]: Yes, added links in the description for the successful builds. Image names have not been affected by this change, only the display name for the tests.

**Review by CooperLink — APPROVED** [review]
Thanks looks great lets try to get it into 4.29.1

**Pull request merged**
